Classes/Controller/CrawlerController.php

```diff
@@ -810,25 +823,36 @@ public function expandParameters($paramArray, $pid)
                             if ($fieldName === 'uid' || $GLOBALS['TCA'][$subpartParams['_TABLE']]['columns'][$fieldName]) {
                                 $queryBuilder = $this->getQueryBuilder($subpartParams['_TABLE']);
 
+                                if($recursiveDepth > 0) {
+                                    /** @var \TYPO3\CMS\Core\Database\QueryGenerator $queryGenerator */
+                                    $queryGenerator = GeneralUtility::makeInstance(\TYPO3\CMS\Core\Database\QueryGenerator::class);
+                                    $pidList = $queryGenerator->getTreeList($lookUpPid, $recursiveDepth, 0, 1);
+                                    $pidArray = GeneralUtility::intExplode(',', $pidList);
+                                } else {
+                                    $pidArray = [(string)$lookUpPid];
+                                }
+                                
                                 $queryBuilder->getRestrictions()
                                     ->removeAll()
                                     ->add(GeneralUtility::makeInstance(DeletedRestriction::class));
 
                                 $queryBuilder
                                     ->select($fieldName)
                                     ->from($subpartParams['_TABLE'])
-                                    // TODO: Check if this works as intended!
-                                    ->add('from', $addTable)
                                     ->where(
-                                        $queryBuilder->expr()->eq($queryBuilder->quoteIdentifier($pidField), $queryBuilder->createNamedParameter($lookUpPid, \PDO::PARAM_INT)),
+                                        $queryBuilder->expr()->in($pidField, $queryBuilder->createNamedParameter($pidArray, Connection::PARAM_INT_ARRAY)),
                                         $where
                                     );
+                                if(!empty($addTable)) {
+                                    // TODO: Check if this works as intended!
+                                    $queryBuilder->add('from', $addTable);
+                                }
                                 $transOrigPointerField = $GLOBALS['TCA'][$subpartParams['_TABLE']]['ctrl']['transOrigPointerField'];
 
                                 if ($subpartParams['_ENABLELANG'] && $transOrigPointerField) {
                                     $queryBuilder->andWhere(
                                         $queryBuilder->expr()->lte(
-                                            $queryBuilder->quoteIdentifier($transOrigPointerField),
```

Comment:
Why is this removed ?

Reply:
because it result in a wrong SQL query. The column name gets "double" quoted.
e.g.:
```suggestion
 [ Doctrine\DBAL\Exception\InvalidFieldNameException ]                                                                           
  An exception occurred while executing 'SELECT `uid` FROM `tx_news_domain_model_news` WHERE (`pid` IN (?, ?, ?)) AND (hidden =   
  0) AND (```l10n_parent``` <= 0) AND (`tx_news_domain_model_news`.`deleted` = 0)' with params [46, 274, 275]:                    
                                                                                                                                  
  Unknown column '`l10n_parent`' in 'where clause'
```